include/session/config/base.hpp

```diff
@@ -917,14 +1001,19 @@ class ConfigBase : public ConfigSig {
     ///   protobuf-wrapped raw message for certain config types).
     ///
     /// Outputs:
-    /// - vector of successfully parsed hashes.  Note that this does not mean the hash was recent or
-    ///   that it changed the config, merely that the returned hash was properly parsed and
-    ///   processed as a config message, even if it was too old to be useful (or was already known
-    ///   to be included).  The hashes will be in the same order as in the input vector.
-    std::vector<std::string> merge(const std::vector<std::pair<std::string, ustring>>& configs);
-
-    // Same as above, but takes values as ustrings (because sometimes that is more convenient).
-    std::vector<std::string> merge(
+    /// - unordered set of successfully parsed hashes.  Note that this does not mean the hash was
+    ///   recent or that it changed the config, merely that the returned hash was properly parsed
+    ///   and processed as a config message, even if it was too old to be useful (or was already
+    ///   known to be included).  For *multipart* message parts that form a complete set of parts
+    ///   (considering all input messages plus any previously stored incomplete parts), the hash
+    ///   will be included if the resulting reconstituted multipart message was a valid config; for
+    ///   parts that do not complete a message set, inclusion in the return value is based only on
+    ///   whether the multipart part itself looked valid.
+    std::unordered_set<std::string> merge(
+            const std::vector<std::pair<std::string, ustring>>& configs);
+
+    // Same as above, but takes values as views (because sometimes that is more convenient).
+    std::unordered_set<std::string> merge(
```

Comment:
I'm sure it'll be fine but we should double check to make sure Desktop & Android can handle getting an `unordered_set` back instead of a `vector`

Reply:
that will be part of the wrapper, and we most likely both will wrap it to a vector/array

Reply:
Yeah there's already a C++ container <-> native container translation that has to happen here.  Depending on how it's written, wrapper code might not even have to change for this one.